libgnucash/engine/gnc-commodity.cpp

```diff
@@ -278,12 +282,16 @@ static gnc_quote_source multiple_quote_sources[] =
     { false, SOURCE_MULTI, "USA (Alphavantage, Fool, ...)", "USA", "usa" },
 };
 
-static const int num_single_quote_sources =
-    sizeof(single_quote_sources) / sizeof(gnc_quote_source);
-static const int num_multiple_quote_sources =
-    sizeof(multiple_quote_sources) / sizeof(gnc_quote_source);
-static std::vector<gnc_quote_source> new_quote_sources;
+static QuoteSourceVec new_quote_sources;
 
+// cannot use map or unordered_map because order must be preserved
```

Comment:
What other part of the code relies on the order to be preserved ?

Reply:
I didn't want to the `gnc_quote_source_lookup_by_internal` search function to modify its behaviour-- it currently searches currency, single, multi, and new quote sources in this order.

Reply:
That's what I guessed and fair enough.

Just fyi you could also have achieved that with an (unordered) map by looping over the source type enum and doing a lookup in the map of the corresponding QuoteSourcesVec associated with each loop iterator. The enum is in the proper order as well. But I don't think it matters so your implementation is fine.

I also briefly thought to reduce the 4 QuoteSourcesVec's to a single one and keeping a separate set of iterators into that vector to indicate start and end of each type. However that's probably something that's best considered in the light of #1626 .